src/main/java/io/mosip/mimoto/controller/PresentationController.java

```diff
@@ -26,31 +26,42 @@ public class PresentationController {
     PresentationService presentationService;
     private final Logger logger = LoggerFactory.getLogger(PresentationController.class);
 
-    @Value("${mosip.inji.ovp.error.redirect.url}")
-    String injiOvpErrorRedirectUrl;
+    @Value("${mosip.inji.ovp.error.redirect.url.pattern}")
+    String injiOvpErrorRedirectUrlPattern;
 
     @Value("${mosip.inji.web.redirect.url}")
     String injiWebRedirectUrl;
 
     @GetMapping("/authorize")
-    public void performAuthorization(HttpServletResponse response, @ModelAttribute PresentationRequestDTO presentationRequestDTO) throws IOException {
+    public void performAuthorization(HttpServletResponse response,
+                                     @RequestParam("response_type") String responseType,
+                                     @RequestParam("resource") String resource,
+                                     @RequestParam("presentation_definition") String presentationDefinition,
+                                     @RequestParam("client_id") String clientId,
+                                     @RequestParam("redirect_uri") String redirectUri ) throws IOException {
         try {
             logger.info("Started Presentation Authorization in the controller.");
+            PresentationRequestDTO presentationRequestDTO = PresentationRequestDTO.builder()
+                    .responseType(responseType)
+                    .resource(resource)
+                    .presentationDefinition(presentationDefinition)
```

Comment:
As per the latest PR, we are taking the request params and building the dto in the controller, can we not serialize to the object in the same place?

Reply:
For Query params, spring boot allows either RequestParams to be used with serialization. or ModelAttribute to Get the queryparams as single object but it does not support serialization from snakecase and camelcase. 
hence we need to go with 
1. having the fields in snake case and use Model Atrribute
2. Have CamelCase in field naming and use RequestMapping.

Hence went ahead with Request Mapping to be consistant with all other objects.

Reply:
My question was more towards presentationDefinition, can we not convert the string to object here itself instead of taking as string and then converting to a object in a different layer ?

Reply:
Addressed and modified it.